drivers/usb/udc/udc_dwc2.c

```diff
@@ -1072,8 +1238,13 @@ static void udc_dwc2_isr_handler(const struct device *dev)
 		LOG_DBG("GINTSTS 0x%x", int_status);
 
 		if (int_status & USB_DWC2_GINTSTS_SOF) {
+			uint32_t dsts;
+
 			/* Clear USB SOF interrupt. */
 			sys_write32(USB_DWC2_GINTSTS_SOF, gintsts_reg);
+
+			dsts = sys_read32((mem_addr_t)&base->dsts);
+			priv->sof_num = usb_dwc2_get_dsts_soffn(dsts);
```

Comment:
Passing the frame number to stack is outside the scope IMHO. It is important to note that the High-Speed SOF repeats the same number in the SOF packet, but here the High-Speed soffn is incremented by one on each of the SOF packets - and therefore if the frame number was passed, it would have to be clear what exactly is supposed to be passed.

The sof number has to be stored due to the even/odd handling necessary for isochronous endpoints.